### DISCUSSION TITLE
cannot load redis server from obj

### DISCUSSION THREAD

**Discussion by as3ii** [rspamd/rspamd#4584] [Q&A]
Hi,
I'm trying to setup various rate limits via rspamd config `/etc/rspamd/local.d/ratelimit.conf`, but when I execute `rspamadm configtest` to check that everything is ok I receive a series of errors like:
```
[...]
cannot load redis server from obj: {[rates] = {[smtp_mail_daily_limit] = {[selector] = user.lower, [bucket] = {[1] = {[rate] = 300 / 24h, [burst] = 300}}}}, [whitelisted_rcpts] = postmaster,mailer-daemon}, processed to {[timeout] = 1, [expand_keys] = false, [read_only] = true}
[...]
syntax OK
```
a similar error appears for other modules, and the rate limit don't works as expected.
Have I messed up something related to redis? Or is it something else?
I allready tried to clean the redis DB with no apparent difference and the problem persists emptying the `local.d/ratelimit.conf` file, so the problem could not be directly related to the ratelimit configs

This server is running debian11, redis 6.0.16, rspamd 3.2 (but the same issue appears on a newer version like 3.6)
The config we tried to deploy:
```
rates {
        smtp_mail_daily_limit = {
          #
          # Default SMTP 300 mail / 24h for authenticated user
          #
          selector = 'user.lower';
          bucket = [
          {
                burst = 300;
                rate = "300 / 24h";
          }]
        }
        web_mail_daily_limit = {
          #
          # Default 20 mail / 24h for NOT authenticated user
          #
          selector = 'digest(header("Subject");header("From"))';
          bucket = [
          {
                burst = 20;
                rate = "20 / 24h";
          }]
        }
}
```

Thanks to everyone that can help me figuring out whats wrong

**Comment by vstakhov** [comment]
It might be related with https://github.com/rspamd/rspamd/pull/4562

**Reply by as3ii** [reply]
if that's the case, what can I do as a temporary fix while I wait for the next release?

(btw the error reported in the issue #4318 is different from the one I encountered, is the cause the same?)
